sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample11_MockingClientTypes.md

```diff
@@ -0,0 +1,392 @@
+
+# Mocking Client Types
+
+Event Hubs is built to support unit testing with mocks, as described in the [Azure SDK Design Guidelines](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-mocking). This is an important feature of the library that allows developers to write tests that are completely focused on their own application logic, though they depend on the Event Hubs types.
+
+
+The examples that follow focus on scenarios likely to occur in applications, and demonstrate how to mock the Event Hubs types typically used in each scenario.
+
+## Publishing Events with the `EventHubProducerClient`
+
+When using batches to publish to Event Hubs, the key interactions with the `EventHubProducerClient` are calling `CreateBatchAsync` to create the batch and `SendAsync` to publish it.   Mocked batches accept a `List<EventData>` that is used as a backing store and can be inspected to verify that the application is adding events to the batch as expected.  The custom `TryAdd` callback can be used to control the decision for whether an event is accepted into the batch or is rejected.   
+
+This snippet demonstrates mocking the `EventHubProducerClient` using Moq, and creating `EventDataBatch` instances using the `EventHubsModelFactory`.
```

Comment:
```suggestion
This snippet demonstrates mocking the `EventHubProducerClient` using Moq and creating `EventDataBatch` instances using the `EventHubsModelFactory`.
```

Reply:
Should we consider linking out to Moq?  I'm on the fence - we don't want to be perceived as endorsing it, but it may confuse people not familiar.   Thoughts?

Reply:
I can include something like "these samples demonstrate mocking using Moq, but many other tools exist that could be used in a similar way" in the intro with a link to it

Reply:
Sounds great!